bindings/python/pybind11.hpp

```diff
@@ -0,0 +1,109 @@
+#include <pybind11/pybind11.h>
+
+#include <pinocchio/multibody/model.hpp>
+
+namespace pinocchio {
+namespace cpp2pybind11 {
+namespace bp = boost::python;
+
+template<typename T> pybind11::object to(T& t)
+{
+  // Create PyObject using boost Python
+  typename bp::reference_existing_object::apply<T*>::type converter;
+  PyObject* pyobj = converter(t);
+  // Create the Pybind11 object
+  return pybind11::reinterpret_borrow<pybind11::object>(pyobj);
+}
+template <typename T>
+auto to(T* t) {
+  // Create PyObject using boost Python
+  typename bp::manage_new_object::apply<T*>::type converter;
+  PyObject* pyobj = converter(t);
+  // Create the Pybind11 object
+  return pybind11::reinterpret_steal<pybind11::object>(pyobj);
+}
+
+template <typename ReturnType>
+ReturnType& from(pybind11::object model) {
+  return boost::python::extract<ReturnType&>(model.ptr());
+}
+
+template <typename T>
+struct convert_type {
+  typedef T type;
+  static inline auto _to(T t) { return t; }
+  static inline auto _from(type t) { return t; }
+};
+template <>
+struct convert_type<void> {
+  static inline void _to() {}
+};
+
+template <typename T>
+struct convert_boost_python_object {
+  typedef pybind11::object type;
+  static inline auto _to(T t) {
+    return to<typename std::remove_pointer<typename std::remove_reference<
+                  typename std::remove_cv<T>::type>::type>::type>(t);
+  }
+  static inline T _from(type t) {
+    return from<
+        typename std::remove_cv<typename std::remove_reference<T>::type>::type>(
+        t);
+  }
+};
+
+#define ADD_CONVERT_TYPE(CLASS)         \
+  template <>                           \
+  struct convert_type<pinocchio::CLASS> \
+      : convert_boost_python_object<pinocchio::CLASS> {}
+
+ADD_CONVERT_TYPE(Model*);
+ADD_CONVERT_TYPE(Model&);
+ADD_CONVERT_TYPE(Model const&);
```

Comment:
```suggestion
ADD_CONVERT_TYPE(Model const&);

undef  ADD_CONVERT_TYPE
```

Reply:
For this macro, it would be nice to change its name and to let it be exposed to the user, as it is a nice shortcut.

Reply:
This is what I intent to do. I don't think it is wise to call `ADD_CONVERT_TYPE` with all possible types here. It is better to let users add conversions for the type they need.

Reply:
By the way, we might also add a file that applies the typedef for most common types.
this should be related to the template instantiation context which is not yet set up. See #1246